src/Microsoft.AspNetCore.Routing/DefaultLinkGenerator.cs

```diff
@@ -317,5 +338,119 @@ public static RouteValueDictionary GetAmbientValues(HttpContext httpContext)
         {
             return httpContext?.Features.Get<IRouteValuesFeature>()?.RouteValues;
         }
+
+        private static class Log
+        {
+            public static class EventIds
+            {
+                public static readonly EventId EndpointsFound = new EventId(100, "EndpointsFound");
+                public static readonly EventId EndpointsNotFound = new EventId(101, "EndpointsNotFound");
+
+                public static readonly EventId TemplateSucceeded = new EventId(102, "TemplateSucceeded");
+                public static readonly EventId TemplateFailedRequiredValues = new EventId(103, "TemplateFailedRequiredValues");
+                public static readonly EventId TemplateFailedConstraint = new EventId(103, "TemplateFailedConstraint");
+                public static readonly EventId TemplateFailedExpansion = new EventId(104, "TemplateFailedExpansion");
+
+                public static readonly EventId LinkGenerationSucceeded = new EventId(105, "LinkGenerationSucceeded");
+                public static readonly EventId LinkGenerationFailed = new EventId(106, "LinkGenerationFailed");
+            }
+
+            private static readonly Action<ILogger, IEnumerable<string>, object, Exception> _endpointsFound = LoggerMessage.Define<IEnumerable<string>, object>(
+                LogLevel.Debug,
+                EventIds.EndpointsFound,
+                "Found the endpoints '{Endpoints}' for address '{Address}'");
```

Comment:
nit: `IEnumerable<string>` is serialized as a comma separated list I am guessing. It would be better if each name was quoted rather than the combined string.

Applicable to all collections in these log messages.

Reply:
Yeah I think you're right about that. Will revise it.

Reply:
I just removed the quoting from this. We don't generally quote things in logs anyway. I added formatting for the route values so that those would show up in a reasonable way.